Update README

johnSchnake opened the pull request:
**What this PR does / why we need it**:
This changes the README in a number of ways:
 - Adds more details about quick mode which is often
used not but front and center in the README
 - Adds improved script lines which take advantate
of --wait options and capturing the filename from
retrieve
 - Adds information regarding `sonobuoy gen` which is the
best workaround whenever Sonobuoy lacks functionality users
need. Slack is usually full of questions where the answer is:
use `sonobuoy gen` and `kubectl apply`
 - Moves the link definitions to a central location at the
bottom of the file.
 - Removes prefix from code lines to improve copy/paste use
 - Clarifies installation can also be from binary which
doesnt require Go and that kubectl isn't required except in
some workflows

**Which issue(s) this PR fixes**
Fixes #614

**Special notes for your reviewer**:

**Release note**:
```
N/A
```

johnSchnake commented on README.md:
```diff
@@ -65,30 +53,49 @@ records the following results:
-[gimme]: https://github.com/travis-ci/gimme
-
-### Download and run
+### Install
```
Should probably list non-go-get installation instructions (just link to our releases/binaries). For some flows this is going to be the more common installation path.

johnSchnake commented on README.md:
````diff
@@ -100,59 +107,57 @@ somewhere local:
 ```
 
-This copies a single `.tar.gz` snapshot from the Sonobuoy pod into your local
+You can then extra the resulting `.tar.gz` snapshot from the Sonobuoy pod into your local
````
s/extra/extract

johnSchnake replied:
Paragraph actually needs reworked; its gotten mangled.

johnSchnake commented on README.md:
```diff
@@ -100,59 +107,57 @@ somewhere local:
+
+# Sonobuoy retrieve echos the created file to stdout so you can capture and manipulate it.
+$ output=$(sonoobuoy retrieve)
+$ mkdir ./results; tar xzf $output -C ./results
```
Side note: I wish github allowed multi-language snippets. I always feel bad listing directions for Mac/Linux without having a Windows version.

johnSchnake commented on README.md:
````diff
@@ -20,104 +38,162 @@ for the following use cases:
+You can check on the status of each of the plugins running with:
 ```
-$ sonobuoy run
+sonobuoy status
````
See https://github.com/civicrm/civicrm-dev-docs/issues/190 for a random discussion of `$` prefix vs not. I think they hit on the relevant points and I liked where they landed.

tl;dr; No prefix unless mixing input/output where you need to distinguish them.

stevesloka replied:
I always put in the prefix and on your issue, they revised to always use: https://github.com/civicrm/civicrm-dev-docs/issues/190#issuecomment-322019933 

I don't feel super strongly either way, but like how putting it in calls out that it's something you run in a terminal.

johnSchnake replied:
Haha, apparently I stopped reading at the first summary when it matched my feelings 😅 I can do either; I wish github markdown had a way to put a non selectable image there instead of text so you could see it but it wouldn’t bother copy paste.

stevesloka replied:
haha, we just need to be consistent. Looks like velero doesn't use them, but I like how it looks when you show a command with output, to distinguish the command from the output. ¯\_(ツ)_/¯

johnSchnake replied:
👍 I'm at least being consistent here; not using them throughout. I only do show single line commands and am not mixing with output so it doesnt really matter in the end now that I've fixed up some other things.

johnSchnake commented:
@stevesloka I originally started with the desire to add details about `--wait` and such but I started finding more things that could use some TLC. Let me know what you think.

Being a readme, it would be great to get some broader community feedback on this too.

stevesloka commented on README.md:
```diff
@@ -1,7 +1,25 @@
 
-[status]: https://travis-ci.org/heptio/sonobuoy.svg?branch=master
-[travis]: https://travis-ci.org/heptio/sonobuoy/#
+- [Overview](#overviewoview)
```
I think the TOC might be too much for a front-page readme. We should get to the point and explain what Sonobuoy is, how it works, and maybe a quick, getting started. I think anything more than that we should move off to other docs pages and dig in deeper.

johnSchnake replied:
TOC removed and readme made shorter

stevesloka commented on README.md:
````diff
@@ -20,104 +38,162 @@ for the following use cases:
+## Getting Started
+To launch Sonobuoy for a short run and wait until it is finished run:
+```
+sonobuoy run --mode quick --wait
````
I think this should remove the `-m quick` and just do the regular scan. That can then merge this section with "checking for Cluster Conformance".

johnSchnake replied:
Did this and removed the quick scan section. Just added a note below this mentioning it as a much shorter alternative for general configuration check.

stevesloka commented on README.md:
```diff
@@ -20,104 +38,162 @@ for the following use cases:
+cluster is conformant according to the [CNCF][cncf].
 
-### Download and run
+### Quick Mode
```
I would ditch this whole section

stevesloka commented on README.md:
```diff
@@ -20,104 +38,162 @@ for the following use cases:
 
-For information on the contents of the snapshot, see the [snapshot
-documentation][snapshot].
+## Advanced use cases
```
I would move this whole section into `docs`

johnSchnake replied:
Did this more or less. Added new docs pages but also linked to them and changed the `more info` section below. I think it looks/flows well and is more likely to get clicks since the topics are more visible and require fewer clicks to get to.

stevesloka commented on README.md:
```diff
@@ -20,104 +38,162 @@ for the following use cases:
 
-This copies a single `.tar.gz` snapshot from the Sonobuoy pod into your local
-`.` directory. Extract the contents into `./results` with:
+There are some Kubernetes e2e tests that may leak some resources. Sonobuoy can
```
Move this to `docs` section. Also, needs to make sure to call out how it does delete with `--all` (i.e. it deletes any namespace starting with `e2e*`)

johnSchnake replied:
Moved to a known issue section and shortened.

stevesloka commented on README.md:
```diff
@@ -20,104 +38,162 @@ for the following use cases:
+
+### Cleanup
+
+Sonobuoy creates a few resources in order to run and expects to run within its
```
Merge this text with the "getting started section"

johnSchnake commented:
@stevesloka I think I addressed all of your concerns, let me know what you think. Since the last review:
 - removed TOC
 - shortened readme
 - moved some of readme into docs pages
 - changed/reworded the more info section
 - a few other minor changes
